pkg/monitor/cluster/hiveregistrationstatus.go

```diff
@@ -31,20 +31,40 @@ func (mon *Monitor) emitHiveRegistrationStatus(ctx context.Context) error {
 		return fmt.Errorf("cluster %s not adopted. No namespace in the clusterdocument", mon.oc.Name)
 	}
 
+	return mon.validateHiveConditions(ctx)
+}
+
+func (mon *Monitor) validateHiveConditions(ctx context.Context) error {
 	cd, err := mon.hiveclientset.HiveV1().ClusterDeployments(mon.oc.Properties.HiveProfile.Namespace).Get(ctx, hive.ClusterDeploymentName, metav1.GetOptions{})
 	if err != nil {
 		return err
 	}
 
+	conditions := mon.filterConditions(ctx, cd, clusterDeploymentConditionsExpected)
```

Comment:
Is it worth passing `clusterDeploymentConditionsExpected` as an arg?

Reply:
It is debatable. I opted for doing to keep the unit test stable in case the list of conditions we are interested in changes.